exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetRecordWriter.java

```diff
@@ -382,4 +396,20 @@ public void cleanup() throws IOException {
 
     codecFactory.release();
   }
+
+  /**
+   * Prepares location where files will be written to.
+   * Creates directory if not present, applies storage strategy.
+   *
+   * @return path to files location
+   * @throws IOException during directory creation or permission setting problems
+   */
+  private Path prepareLocationPath() throws IOException {
+    if (locationPath == null) {
+      locationPath = new Path(location);
+      fs.mkdirs(locationPath);
+      storageStrategy.apply(fs, locationPath);
```

Comment:
This will apply the permissions only to the deepest of the directories created by mkdirs?